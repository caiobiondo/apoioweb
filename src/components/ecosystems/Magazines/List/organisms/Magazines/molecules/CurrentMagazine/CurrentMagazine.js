import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import { translate } from 'locale';
import { FlatButton, Icon, FormButton } from 'natura-ui';
import { Dialog } from 'material-ui';
import {
  CurrentMagazineWrapper,
  CurrentMagazineHeader,
  CurrentMagazineCover,
  CurrentMagazinePeriod,
  CurrentMagazineTitle,
  CurrentMagazineDescription,
  CurrentMagazineTaxWrapper,
  CurrentMagazineTaxInfoTitle,
  CurrentMagazineTaxInfoDescription,
  CurrentMagazineTaxViewMore,
  ButtonWrapper,
  CurrentMagazineInfo,
  CurrentMagazineInfoWrapper,
  CurrentMagazineAdditionalInfo,
  CurrentMagazineSeeMore,
  CurrentMagazineTaxItem,
  CurrentMagazineTaxItemDatum,
  CurrentMagazineHeaderTaxItemDatum,
} from './CurrentMagazine.styles';
import ImageWithFallback from 'components/molecules/ImageWithFallback';
import { FormattedNumber } from 'react-intl';

export class CurrentMagazine extends Component {
  state = {
    additionalInfoOpened: false,
    taxViewMoreOpened: false,
  };

  downloadMagazine = () => {
    const { magazine } = this.props;
    window.open(magazine.pdfFile);
  };

  openMagazine = () => {
    const { magazine, type } = this.props;
    this.props.history.push(`/magazines/view/${type}/${magazine.id}`);
  };

  toggleAdditionalInfo = () => {
    this.setState({ additionalInfoOpened: !this.state.additionalInfoOpened });
  };

  openTaxViewMore = () => {
    this.setState({ taxViewMoreOpened: !this.state.taxViewMoreOpened });
  };

  handleTaxViewMoreClose = () => {
    this.setState({ taxViewMoreOpened: false });
  };

  renderTaxViewMoreActions = () => {
    return [
      <FlatButton label={translate('ok')} primary={true} onClick={this.handleTaxViewMoreClose} />,
    ];
  };

  renderTaxViewMoreDialog = () => {
    const dialogData = [
      { productType: translate('beardProducts'), tax: 56.48 },
      { productType: translate('hairProducts'), tax: 51.33 },
      { productType: translate('bodyProducts'), tax: 47.03 },
      { productType: translate('deodorantColony'), tax: 44.99 },
      { productType: translate('deodorantAnti'), tax: 44.99 },
      { productType: translate('makeupProducts'), tax: 55.94 },
      { productType: translate('bodyOils'), tax: 45.78 },
      { productType: translate('parfum'), tax: 49.16 },
    ];

    return (
      <Dialog
        title={translate('taxBurden')}
        actions={this.renderTaxViewMoreActions()}
        open={this.state.taxViewMoreOpened}
        modal={false}
        onRequestClose={this.handleTaxViewMoreClose}
      >
        <CurrentMagazineTaxItem>
          <CurrentMagazineHeaderTaxItemDatum>
            {translate('productType')}
          </CurrentMagazineHeaderTaxItemDatum>
          <CurrentMagazineHeaderTaxItemDatum>
            {translate('taxIBPT')}
          </CurrentMagazineHeaderTaxItemDatum>
        </CurrentMagazineTaxItem>
        {dialogData.map((datum, index) => {
          return (
            <CurrentMagazineTaxItem key={index}>
              <CurrentMagazineTaxItemDatum>{datum.productType}</CurrentMagazineTaxItemDatum>
              <CurrentMagazineTaxItemDatum>
                <FormattedNumber value={datum.tax} />%
              </CurrentMagazineTaxItemDatum>
            </CurrentMagazineTaxItem>
          );
        })}
      </Dialog>
    );
  };

  render() {
    const { magazine } = this.props;
    const { additionalInfoOpened } = this.state;

    if (!magazine) return null;

    return (
      <CurrentMagazineWrapper>
        <CurrentMagazineHeader>{translate('currentMagazine')}</CurrentMagazineHeader>
        <CurrentMagazineInfoWrapper>
          <CurrentMagazineCover>
            <ImageWithFallback imageUrl={magazine.highlightImage} />
          </CurrentMagazineCover>
          <CurrentMagazineInfo>
            <CurrentMagazinePeriod>
              {translate('magazineCycle')} {magazine.period}
            </CurrentMagazinePeriod>
            <CurrentMagazineTitle>{magazine.title}</CurrentMagazineTitle>
            <CurrentMagazineAdditionalInfo opened={additionalInfoOpened}>
              <CurrentMagazineDescription
                dangerouslySetInnerHTML={{ __html: magazine.description }}
              />
              <ButtonWrapper>
                <FlatButton
                  label={translate('downloadMagazine')}
                  primary
                  onClick={this.downloadMagazine}
                  icon={<Icon file="ico_magazine_download" />}
                />
                <FlatButton
                  label={translate('visualizeMagazine')}
                  primary
                  onClick={this.openMagazine}
                  icon={<Icon file="ico_view" />}
                />
              </ButtonWrapper>
              <CurrentMagazineTaxWrapper>
                <CurrentMagazineTaxInfoTitle>
                  <CurrentMagazineTaxViewMore onClick={this.openTaxViewMore}>
                    {translate('taxViewMore')}
                  </CurrentMagazineTaxViewMore>
                  {translate('taxInfoTitle')}
                </CurrentMagazineTaxInfoTitle>
                <CurrentMagazineTaxInfoDescription>
                  {translate('taxInfoDescription')}
                </CurrentMagazineTaxInfoDescription>
              </CurrentMagazineTaxWrapper>
            </CurrentMagazineAdditionalInfo>
            <CurrentMagazineSeeMore>
              <FormButton
                fullWidth
                primary
                label={additionalInfoOpened ? translate('hide') : translate('viewMore')}
                onClick={this.toggleAdditionalInfo}
                raised="true"
              />
            </CurrentMagazineSeeMore>
          </CurrentMagazineInfo>
        </CurrentMagazineInfoWrapper>
        {this.renderTaxViewMoreDialog()}
      </CurrentMagazineWrapper>
    );
  }
}

export default withRouter(CurrentMagazine);
