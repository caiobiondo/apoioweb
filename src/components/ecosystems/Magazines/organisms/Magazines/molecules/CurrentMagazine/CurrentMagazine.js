import React from 'react';
import { translate } from 'locale';
import { FlatButton, Icon } from 'natura-ui';

export default ({ magazine }) => {
  console.log(magazine);
  return (
    <div>
      <h2>{translate('currentMagazine')}</h2>
      <img src={magazine.highlightImage} alt={translate('currentMagazine')} />
      <span>
        {translate('magazineCycle')} {magazine.period}
      </span>
      <h3>{magazine.title}</h3>
      <p dangerouslySetInnerHTML={{ __html: magazine.description }} />
      <p>{translate('taxInfoTitle')}</p>
      <p>{translate('taxInfoDescription')}</p>
      <FlatButton label={translate('visualizeMagazine')} primary icon={<Icon file="ico_view" />} />
      <FlatButton
        label={translate('downloadMagazine')}
        primary
        icon={<Icon file="ico_magazine_download" />}
      />
    </div>
  );
};
