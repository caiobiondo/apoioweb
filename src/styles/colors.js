//
// Color Specs
// -------------------------------------
export const black = '#000';
export const white = '#fff';

export const gray100 = '#eeeeee';
export const gray150 = '#dddddd';
export const gray200 = '#bbbbbb';
export const gray300 = '#8e8e8e';
export const gray600 = '#606060';
export const gray890 = '#333333';
export const gray900 = '#222222';

export const orange100 = '#f4a652';
