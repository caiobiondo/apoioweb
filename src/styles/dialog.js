export const dialogContainer = {
  maxWidth: 'none',
};

export const dialogContent = {
  fontFamily: 'Roboto',
  padding: '12px 25px',
};

export const dialogTitle = {
  fontWeight: 800,
};

export const dialogActions = {
  fontFamily: 'Roboto',
};
