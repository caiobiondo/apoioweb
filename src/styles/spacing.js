//
// Spacing
// -------------------------------------
export const spSmall = '5px';
export const spMedium = '24px';
export const spPage = '40px';

export const screenXs = '576px';
export const screenSm = '576px';
export const screenMd = '768px';
export const screenLg = '992px';
export const screenXl = '1200px';
