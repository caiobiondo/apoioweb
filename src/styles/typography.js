//
// Font Weights
// -------------------------------------
export const fw300 = '300';
export const fw400 = '400';
export const fw600 = '600';
export const fw700 = '700';
export const fw900 = '900';

//
// Font Sizes
// -------------------------------------
export const fs9 = '9px';
export const fs10 = '10px';
export const fs12 = '12px';
export const fs13 = '13px';
export const fs14 = '14px';
export const fs15 = '15px';
export const fs16 = '16px';
export const fs18 = '18px';
export const fs20 = '20px';
export const fs25 = '25px';
export const fs28 = '28px';
export const fs32 = '32px';

//
// Font Families
//

export const NaturaBold = 'NaturaSans-Bold';
export const NaturaRegular = 'NaturaSans-Regular';
export const RobotoRegular = 'Roboto-Regular';
export const RobotoMedium = 'Roboto-Medium';
export const RobotoLight = 'Roboto-Light';
