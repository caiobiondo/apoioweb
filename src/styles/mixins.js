import styled, { css } from 'styled-components';
import { Responsive } from '@entria/components';
import { spMedium } from 'styles/spacing';

export const Full = css`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    top: 0;
    left: ${spMedium};
    right: ${spMedium};
    bottom: 0;
  }
`;

export const Center = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
`;

export const LoadingWrapperStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  margin-top: 40px;
`;
