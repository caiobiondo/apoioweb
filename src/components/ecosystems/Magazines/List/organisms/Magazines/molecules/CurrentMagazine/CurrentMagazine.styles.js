import styled from 'styled-components';
import { Responsive } from '@entria/components';
import { spMedium } from 'styles/spacing';
import { orange100, gray100, gray700 } from 'styles/colors';
import { NaturaRegular, RobotoLight, RobotoRegular, RobotoMedium, fw600 } from 'styles/typography';

export const CurrentMagazineWrapper = styled.div`
  padding: ${spMedium};
`;

export const CurrentMagazineHeader = styled.h2`
  font-family: ${RobotoLight};
  font-size: 21px;
  font-weight: 300;
  line-height: 1.19;
  text-align: left;
  color: ${orange100};
`;

export const CurrentMagazineInfoWrapper = styled.div`
  display: flex;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    flex-direction: column;
  }
`;

export const CurrentMagazineInfo = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1 1 0%;
  padding-left: 62.5px;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    display: block;
    padding: 20px;
  }
`;

export const CurrentMagazineCover = styled.div`
  width: 325.5px;
  height: 444px;
  position: relative;

  > div {
    height: 100%;
  }

  img {
    height: 100%;
    object-fit: contain;
    box-shadow: 1.6px 1.9px 7.5px 0 rgba(0, 0, 0, 0.1);

    @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
      align-self: center;
    }
  }
`;

export const CurrentMagazinePeriod = styled.span`
  font-family: ${RobotoMedium};
  font-size: 12.7px;
  font-weight: bold;
  line-height: 0.71;
  text-align: left;
  text-transform: uppercase;
  color: ${gray700};
`;

export const CurrentMagazineTitle = styled.h3`
  font-family: ${NaturaRegular};
  font-size: 26.6px;
  font-weight: bold;
  line-height: 1.18;
  text-align: left;
  color: ${orange100};
  margin: 10.5px 0 0 0;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    font-size: 18px;
    overflow: hidden;
  }
`;

export const CurrentMagazineDescription = styled.p`
  font-family: ${RobotoRegular};
  font-size: 19.5px;
  line-height: 1.28;
  text-align: left;
  color: ${gray700};
  padding: 25px 0;
  margin: 0;

  p {
    margin: 0;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    font-size: 13px;
    padding: 10px 0;
    text-align: center;
  }
`;

export const CurrentMagazineTax = styled.p`
  font-family: ${RobotoRegular};
  font-size: 10px;
  line-height: 1.25;
  text-align: left;
  color: ${gray700};
  margin: 0;
`;

export const CurrentMagazineTaxInfoTitle = CurrentMagazineTax.extend`
  font-size: 11.5px;
  line-height: 2.17;
  padding-bottom: 10px;
`;

export const CurrentMagazineTaxViewMore = styled.span`
  color: ${gray700};
  font-family: ${RobotoMedium};
  font-weight: bold;
  text-decoration: none;
  margin-right: 5px;
  cursor: pointer;
`;

export const CurrentMagazineTaxInfoDescription = CurrentMagazineTax.extend`
  font-size: 11.5px;
  line-height: 2.17;
`;

export const ButtonWrapper = styled.div`
  flex: 0 1 auto;

  svg {
    fill: ${orange100};
    width: 13px;
    padding-left: 16px;
  }

  span {
    font-family: ${RobotoRegular};
    font-size: 16.9px;
    font-weight: 500;
    line-height: 1.28;
    text-align: left;
    color: ${orange100};
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    align-items: center;
    margin-top: 5px;

    button {
      width: 100%;
    }
  }
`;

export const CurrentMagazineTaxWrapper = styled.div`
  flex: 1 1 0%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    display: none;
  }
`;

export const CurrentMagazineAdditionalInfo = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1 1 0%;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    ${props => {
      return props.opened ? 'height: auto; opacity: 1;' : 'height: 0; opacity: 0;';
    }};
  }
`;

export const CurrentMagazineSeeMore = styled.div`
  display: none;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    display: block;
    margin: 10px 0;
  }
`;

export const CurrentMagazineTaxItem = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 8px 0 5px 0;
  color: ${gray700};

  & + & {
    border-top: 1px solid ${gray100};
  }
`;

export const CurrentMagazineTaxItemDatum = styled.div`
  line-height: 1.25;
`;

export const CurrentMagazineHeaderTaxItemDatum = styled.div`
  font-weight: ${fw600};
`;
