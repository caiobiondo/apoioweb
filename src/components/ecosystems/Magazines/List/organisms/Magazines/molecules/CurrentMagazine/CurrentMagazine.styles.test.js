import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { CurrentMagazineAdditionalInfo } from './CurrentMagazine.styles';

describe('CurrentMagazine Styles', () => {
  it('should render additional info opened (visible)', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<CurrentMagazineAdditionalInfo opened={true} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render additional info not opened (not visible)', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<CurrentMagazineAdditionalInfo opened={false} />);
    expect(tree).toMatchSnapshot();
  });
});
