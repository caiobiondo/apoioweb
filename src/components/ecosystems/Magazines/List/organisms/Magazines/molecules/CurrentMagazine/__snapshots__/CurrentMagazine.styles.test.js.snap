// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentMagazine Styles should render additional info not opened (not visible) 1`] = `
<div
  className="CurrentMagazinestyles__CurrentMagazineAdditionalInfo-hgzQrm eLcuzc"
/>
`;

exports[`CurrentMagazine Styles should render additional info opened (visible) 1`] = `
<div
  className="CurrentMagazinestyles__CurrentMagazineAdditionalInfo-hgzQrm fiRrFU"
/>
`;
